iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java

```diff
@@ -94,4 +101,58 @@ public TimeValuePairSorter query(String deviceId, String measurement, TSDataType
     return memTableMap.get(deviceId).get(measurement);
   }
 
+  @Override
+  public void delele(String deviceId, String measurementId, long timestamp) {
+    Map<String, IWritableMemChunk> deviceMap = memTableMap.get(deviceId);
+    if (deviceMap != null) {
+      IWritableMemChunk chunk = deviceMap.get(measurementId);
+      IWritableMemChunk newChunk = filterChunk(chunk, timestamp);
+      if (newChunk != null) {
+        deviceMap.put(measurementId, newChunk);
+      }
+    }
+  }
+
+  /**
+   * If chunk contains data with timestamp less than 'timestamp', create a copy and delete all
+   * those data. Otherwise return null.
+   * @param chunk the source chunk.
+   * @param timestamp the upper-bound of deletion time.
+   * @return A reduced copy of chunk if chunk contains data with timestamp less than 'timestamp',
+   * of null.
+   */
+  private IWritableMemChunk filterChunk(IWritableMemChunk chunk, long timestamp) {
```

Comment:
how about using the concept of `Page`?
Suppose a Chunk consists of page1, page2, page3,...., page n, and you will delete page 1, 2 and a part of page 3.
Then, just re-compress page 3 (and as a result, this page will have just a few points), and keep page 4~n into the new Chunk.

Reply:
Good idea, but unfortunately IWritableMemChunk does not provide such interfaces. I would suggest you issue this and improve it later.